stl/inc/syncstream

```diff
@@ -0,0 +1,329 @@
+// syncstream standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _SYNCSTREAM_
+#define _SYNCSTREAM_
+#include <yvals.h>
+#if _STL_COMPILER_PREPROCESSOR
+#if !_HAS_CXX20
+#pragma message("The contents of <syncstream> are available only with C++20 or later.")
+#else // ^^^ !_HAS_CXX20 / _HAS_CXX20 vvv
+#include <memory>
+#include <ostream>
+#include <shared_mutex>
+#include <streambuf>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#pragma push_macro("emit")
+#undef new
+#undef emit
+
+_STD_BEGIN
+
+extern "C" shared_mutex& __cdecl _Get_mutex_for_instance(void* _Ptr);
+extern "C" void __cdecl _Acquire_mutex_for_instance(void* _Ptr) noexcept;
+extern "C" void __cdecl _Release_mutex_for_instance(void* _Ptr) noexcept;
+
+// CLASS TEMPLATE basic_syncbuf
+template <class _Elem, class _Traits, class _Alloc>
+class basic_syncbuf : public basic_streambuf<_Elem, _Traits> {
+public:
+    using int_type       = typename _Traits::int_type;
+    using pos_type       = typename _Traits::pos_type;
+    using off_type       = typename _Traits::off_type;
+    using allocator_type = _Alloc;
+    using streambuf_type = basic_streambuf<_Elem, _Traits>;
+
+    using pointer   = typename allocator_traits<_Alloc>::pointer;
+    using size_type = typename allocator_traits<_Alloc>::size_type;
+
+    basic_syncbuf() = default;
+
+    explicit basic_syncbuf(streambuf_type* _Strbuf) : basic_syncbuf(_Strbuf, _Alloc{}) {}
+
+    basic_syncbuf(streambuf_type* _Strbuf, const _Alloc& _Al_) : _Wrapped(_Strbuf), _Al(_Al_) {
+        if (_Wrapped != nullptr) {

```

Comment:
I think that this check could be beneficial, in case an object was created explicitly with `nullptr`. The internal map will hold a non-necessary `mutex` for the program's lifetime since there's no way to erase it.

Reply:
I think he meant to remove `!= nullptr` and stay with `if (_Wrapped)`